packages/integrations/tailwind/src/v4.ts

```diff
@@ -0,0 +1,37 @@
+import tailwindcss from '@tailwindcss/vite';
+import type { AstroIntegration } from 'astro';
+
+type TailwindOptions = {
+	/**
+	 * Apply Tailwind's base styles
+	 * Disabling this is useful when further customization of Tailwind styles
+	 * and directives is required. See {@link https://tailwindcss.com/docs/functions-and-directives#tailwind Tailwind's docs}
+	 * for more details on directives and customization.
+	 * @default true
+	 */
+	applyBaseStyles?: boolean;
+};
+
+export default function tailwindIntegration(options?: TailwindOptions): AstroIntegration {
+	const applyBaseStyles = options?.applyBaseStyles ?? true;
+
+	return {
+		name: '@astrojs/tailwind',
+		hooks: {
+			'astro:config:setup': async ({ updateConfig, injectScript }) => {
+				// Inject the Tailwind vite plugin
+				updateConfig({
+					vite: {
+						plugins: [tailwindcss()],
+						css: { transformer: 'lightningcss' },
+					},
+				});
+
+				if (applyBaseStyles) {
+					// Inject the Tailwind base import
+					injectScript('page-ssr', `import '@astrojs/tailwind/base-v4.css';`);
```

Comment:
Couldn't this be problematic, because it would inject the import onto every page?

The import should only happen on the most top level of any given page. So if you have a layout the import should happen inside the layout and not the page.

I'm unsure if vite optimizes that, but I don't think it does.

Reply:
Well vite does optimise by default unless you specify to not minify not only that you can also have the ability to disable it by doing applybasestyles to false.

Reply:
Yeah. However that is not the point, I tried to make. Just because it currently is build that way it doesn't mean it is the best solution. However I really haven't tested if it has an impact - if the global stylesheet is imported on every page, instead of just once.

Just my personalfeeling but I guess it does matter, and I personally would mind. And if it does I would at least expect an note about that in docs, so users are aware of it. 

But I'll leave it with that. :)

Reply:
Well with v4 they are now using @import "tailwindcss" instead of `@tailwind` so it has to be separate since v4 is different to v3.

Reply:
I felt like this change is completely useless since V4 has different stuff now. You Check is out [here](https://tailwindcss.com/blog/tailwindcss-v4-alpha)

Reply:
And also a lot of people don't want to apply the base styles so we give them the choice of that. So Really unnecessary and not best pratices to use the same css file when v4 uses import.